src/tree.rs

```diff
@@ -125,7 +125,7 @@ impl<'tree> TreeEntry<'tree> {
     ///
     /// The lifetime of the entry is tied to the tree provided and the function
     /// is unsafe because the validity of the pointer cannot be guaranteed.
-    unsafe fn from_raw_const(raw: *const raw::git_tree_entry)
+    pub unsafe fn from_raw_const(raw: *const raw::git_tree_entry)
```

Comment:
I'd prefer to leave this private for now (inaccessible to external users of the crate), could this be made a public free function in this module?